File: task-1/src/main/java/ru/biluta/agents/PongAgent.java
package ru.biluta.agents;

import jade.core.AID;
import jade.core.Agent;
import jade.core.behaviours.CyclicBehaviour;
import jade.core.behaviours.OneShotBehaviour;
import jade.lang.acl.ACLMessage;
import jade.lang.acl.MessageTemplate;

public class PongAgent extends Agent {
    @Override
    protected void setup() {
        addBehaviour(new CyclicBehaviour() {
            @Override
            public void action() {
                ACLMessage msg = receive(MessageTemplate.MatchContent("ping"));
                if (msg != null) {
                    System.out.println("Пинг получен " + msg.getSender().getLocalName());
                    ACLMessage reply = msg.createReply();
                    reply.setContent("pong");
                    send(reply);

                    System.out.println("Понг отправлен " + msg.getSender().getLocalName());
                } else {
                    block();
                }
            }
        });
    }
}

File: task-1-kotlin/src/main/kotlin/agents/PongAgent.kt
package agents

import jade.core.Agent
import jade.core.behaviours.CyclicBehaviour
import jade.lang.acl.ACLMessage

class PongAgent : Agent() {
    override fun setup() {
        println("Привет! Агент Pong ${aid.name} готов.")
        addBehaviour(PongBehaviour())
    }

    inner class PongBehaviour : CyclicBehaviour() {
        override fun action() {
            val msg = receive()
            if (msg != null && msg.content == "ping") {
                val reply = msg.createReply()
                reply.content = "pong"
                send(reply)
                println("Pong sent by ${aid.name}")
            }
            block()
        }
    }

    override fun takeDown() {
        println("Агент Pong ${aid.name} завершает работу.")
    }
}
